### PR TITLE
Adding serial console

### PR DESCRIPTION
**Pull request by metalice**
## 📝 Description

Serial console + actions + console selector dropdown

Based on https://github.com/kubevirt-ui/kubevirt-plugin/pull/120 - should be merged first

## 🎥 Demo

> Please add a video or an image of the behavior/changes

### CODE REVIEW

File: src/views/virtualmachinesinstance/details/tabs/console/components/AccessConsoles/AccessConsoles.tsx [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+      } else {
+        const typeText = getChildTypeName(child);
+        acc.push(
+          <SelectOption
+            key={typeText}
+            id={typeText}
+            value={{ value: typeText, toString: () => typeText } as SelectOptionObject}
+          />,
+        );
+      }
```
Comment (upalatucci) [inline src/views/virtualmachinesinstance/details/tabs/console/components/AccessConsoles/AccessConsoles.tsx:92]: Can this be done like this?
```js
        const typeText = getChildTypeName(child);
        acc.push(
          <SelectOption
            key={typeText}
            id={typeText}
            value={{ value: typeText, toString: () => typeMap[typeText] || typeText } as SelectOptionObject}
          />,
        );
```

**Review by upalatucci — COMMENTED** [review]
Don't know if this can help :-) Little suggestion

File: src/views/virtualmachinesinstance/details/tabs/console/components/AccessConsoles/AccessConsoles.tsx (hunk 1 above)
Comment (metalice) [reply src/views/virtualmachinesinstance/details/tabs/console/components/AccessConsoles/AccessConsoles.tsx]: great suggestion

Comment (yaacov) [reply src/views/virtualmachinesinstance/details/tabs/console/components/AccessConsoles/AccessConsoles.tsx]: why do we maintain a copy of patternflys component?
https://github.com/patternfly/patternfly-react/blob/main/packages/react-console/src/components/AccessConsoles/AccessConsoles.tsx

if we need to add new features should we add them in patternfly?
if we actually want to maintain our own version, we should do it in a more public place, maybe notify the patently guys we are doing it?

Comment (metalice) [reply src/views/virtualmachinesinstance/details/tabs/console/components/AccessConsoles/AccessConsoles.tsx]: If we will need a new feature we can simply add it to our own code base, pattternfly/console codebase isn't up-to-date with the rest of the console code and causes conflicts. 
If we will want the far future to do it in a public place we can think over the approach.

**Pull request merged**